core/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -296,6 +312,28 @@ public static SearchResponse fromXContent(XContentParser parser) throws IOExcept
                             parser.skipChildren();
                         }
                     }
+                } else if (Clusters._CLUSTERS_FIELD.match(currentFieldName)) {
+                    int successful = -1;
+                    int total = -1;
+                    int skipped = -1;
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                        if (token == XContentParser.Token.FIELD_NAME) {
+                            currentFieldName = parser.currentName();
+                        } else if (token.isValue()) {
+                            if (Clusters.SUCCESSFUL_FIELD.match(currentFieldName)) {
+                                successful = parser.intValue();
+                            } else if (Clusters.TOTAL_FIELD.match(currentFieldName)) {
+                                total = parser.intValue();
+                            } else if (Clusters.SKIPPED_FIELD.match(currentFieldName)) {
+                                skipped = parser.intValue();
+                            } else {
+                                parser.skipChildren();
```

Comment:
should we be more strict and just fail if this happens?

Reply:
this parsing method is used in the high-level REST client, we are lenient there to guarantee forward compatibility, meaning that if one day we add a new field under _clusters, we don't break while parsing that but we rather ignore it and everything is fine. This is tested in SearchResponseTests#testFromXContentWithRandomFields

Reply:
got it